File: viewpagerwithoutfragment/src/main/java/com/kirchhoff/viewpagerwithoutfragment/VPWithoutFragmentActivity.java
package com.kirchhoff.viewpagerwithoutfragment;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;


/**
 * @author Kirchhoff-
 */

public class VPWithoutFragmentActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.a_vp_without_fragment);

        ViewPager viewPager = (ViewPager) findViewById(R.id.viewpager);
        viewPager.setAdapter(new CustomPagerAdapter(this));
    }
}

File: viewpagerwithoutfragment/src/main/java/com/kirchhoff/viewpagerwithoutfragment/VPWithoutFragmentActivity.kt
package com.kirchhoff.viewpagerwithoutfragment

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager

class VPWithoutFragmentActivity : AppCompatActivity(R.layout.a_vp_without_fragment) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val viewPager = findViewById<ViewPager>(R.id.viewpager)
        viewPager.adapter = CustomPagerAdapter(this)
    }
}
